cmd/update-main.go

```diff
@@ -235,20 +256,33 @@ func getDownloadURL() (downloadURL string) {
 	return minioReleaseURL + "minio"
 }
 
-func getUpdateInfo(timeout time.Duration, mode string) (older time.Duration, downloadURL string, err error) {
-	currentReleaseTime, err := GetCurrentReleaseTime()
+// getUpdateInfo - compares latest available official release with the
+// version of current server process; returns staleness of current
+// release and download URL of latest release.
+func getUpdateInfo(mode string, timeout time.Duration) (
+	older time.Duration, downloadURL string, err error) {
+
+	u := updateCheckParams{
+		version:            Version,
+		serverBinaryPath:   os.Args[0],
+		cgroupFile:         "/proc/self/cgroup",
+		releaseChecksumURL: minioReleaseURL + "minio.shasum",
+		serverMode:         mode,
+	}
```

Comment:
There are two options:

1) Passing variables as function parameters so that the functions are unit testable. Downside is longer function signature (or use structs like updateCheckParams and have methods on it)

2) Use global variables, harder to unit test but upside is simpler function signature.

We use 2nd style in minio all over, let's discuss what style we should follow so that we can be consistent.